test/Tests/src/Semantic/Warnings_Spec.enso

```diff
@@ -388,5 +388,19 @@ spec = Test.group "Dataflow Warnings" <|
         Problems.assume_no_problems <| condition_1 False x3
         Problems.assume_no_problems <| condition_2 False x3
 
+    Test.specify "should only report unique warnings" <|
+        a = 1
+        b = Warning.attach "Foo!" a
+        c = Warning.attach "Bar!" b
+        d = Warning.attach "Foo!" b
+
+        result_1 = b + c
+        Warning.get_all result_1 . map (x-> x.value.to_text) . should_equal ["Bar!", "Foo!"]
+
+        result_2 = b + b + b
+        Warning.get_all result_2 . length . should_equal 1
+
+        result_3 = b + b + d
+        Warning.get_all result_3 . map (x-> x.value.to_text) . should_equal ["Foo!", "Foo!"]
```

Comment:
This shows that the _warning message_ isn't important - two warnings can have the same message and they will be treated as different. OK.

Reply:
Just a note that the warnings don't really have a _message_ but a generic payload which can be any object (in these examples they are just `Text` payloads).